### ISSUE TITLE
Installing plugins over ssh gives error E440

### ISSUE THREAD

**Issue by edi9999** [junegunn/vim-plug#499]
I'm trying to set up a dev server with some scripts to set them up, and at the end I'm doing 

```
ssh myip nvim +PlugInstall +qall
```

Here's a minimal vimrc to reproduce the problem : 

```
set nocompatible

syntax enable

call plug#begin()
Plug 'rking/ag.vim', {'on': 'Ag'}
Plug 'vim-scripts/ReplaceWithRegister'
call plug#end()
```

---

```
NVIM 0.1.5-dev
Build type: RelWithDebInfo
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector-strong -Wformat -Werror=fo
rmat-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DDISABLE_LOG -Wall -W
extra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-s
trong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -D_GNU_SOURCE
-I/build/neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235+2620+20~ubuntu16.04.1/build/config
-I/build/neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235+2620+20~ubuntu16.04.1/src -I/build/
neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235+2620+20~ubuntu16.04.1/.deps/usr/include -I/b
uild/neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235+2620+20~ubuntu16.04.1/.deps/usr/include
 -I/build/neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235+2620+20~ubuntu16.04.1/.deps/usr/in
clude -I/build/neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235+2620+20~ubuntu16.04.1/.deps/u
sr/include -I/build/neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235+2620+20~ubuntu16.04.1/.d
eps/usr/include -I/build/neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235+2620+20~ubuntu16.04
.1/.deps/usr/include -I/usr/include -I/build/neovim-GvBvnu/neovim-0.1.4ubuntu1+git201606301235
+2620+20~ubuntu16.04.1/build/src/nvim/auto -I/build/neovim-GvBvnu/neovim-0.1.4ubuntu1+git20160
6301235+2620+20~ubuntu16.04.1/build/include
Compiled by buildd@lgw01-22




Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui
For differences from Vim, see :help vim-differences




   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

This works fine on vim, but doesn't work on neovim.

Here's the full error log : 

```
nvim +PlugInstall +qall
Vim: Warning: Output is not to a terminal
Vim: Warning: Input is not from a terminal
"[Plugins]" [New File]
Error detected while processing function <SNR>7_job_handler[16]..<SNR>7_log[2]..<SNR>7_logpos:
line    1:
E727: Start past end
E714: List required
Error detected while processing function <SNR>7_job_handler[23]..<SNR>7_reap[9]..<SNR>7_log[2]..<SNR>7_logpos:
line    1:
E727: Start past end
E714: List required
Error detected while processing function <SNR>7_job_handler[23]..<SNR>7_reap[10]..<SNR>7_bar:
line    3:
E440: undo line missing
Error detected while processing function <SNR>7_job_handler[23]..<SNR>7_reap[9]..<SNR>7_log[2]..<SNR>7_logpos:
line    1:
E727: Start past end
E714: List required
Error detected while processing function <SNR>7_job_handler[24]..<SNR>7_tick[7]..<SNR>7_update_finish[7]..<SNR>7_logpos:
line    1:
E727: Start past end
E714: List required
E727: Start past end
E714: List required
Error detected while processing function <SNR>7_job_handler[24]..<SNR>7_tick[7]..<SNR>7_update_finish:
line   48:
E16: Invalid range:     4 d _
```

I found out that the code of ReplaceRegister isn't even cloned, so I don't know exactly what is happening.
- Type:
  - [x] Bug
  - [ ] Enhancement
  - [ ] Feature Request
  - [ ] Question
- OS:
  - [ ] All/Other
  - [x] Linux
  - [ ] OS X
  - [ ] Windows
- Vim:
  - [ ] Terminal Vim
  - [ ] GVim
  - [x] Neovim

Let me know if you need additionnal info.

The devserver is Ubuntu 16.04, but I don't think it has anything to do with the used server

**Comment by edi9999** [comment]
The error log is rather just : 

```
"[Plugins]" [New File]
Error detected while processing function <SNR>7_job_handler[23]..<SNR>7_reap[10]..<SNR>7_bar:
line    3:
E440: undo line missing
```

**Comment by edi9999** [comment]
It seems that it happens because of the `+qall`.

The running async jobs are "killed" because of the `+qall`.

Maybe they is a way of waiting to download all plugins before exiting

**Comment by edi9999** [comment]
Probably related to https://github.com/junegunn/vim-plug/issues/104

**Comment by edi9999** [comment]
I had has("python") set to 0.

Installing neovim python with 

pip install neovim solves the issue, but i think still that this is a bug when you have not vim python

**Comment by junegunn** [comment]
#104 is an open issue we haven't been able to fix yet.
- Choosing Python installer during `vim_starting` is an effort to address the issue, but like you have noticed, it doesn't always help.
- We can't just fall back to single-threaded blocking installer during `vim_starting` as we can't be sure of the intention of the user. I mean doing so makes sense for `nvim +PlugUpdate +qa`, but what if the user doesn't intend to quit after the installer finishes? `nvim +PlugUpdate`
- Last time I checked, neovim exited without error messages even though the installer failed to finish its job, but with the Iatest version of neovim, I can reproduce the error messages you posted. Not sure what has changed. (We don't see such errors when we manually interrupt the installer by pressing `q`.)
- So basically this is a duplicate of #104.
- A very easy workaround to this particular use case is to use ordinary vim instead of neovim. You'll have no problem with `vim +PlugUpdate +qa` as the installer in Vim is blocking.

**Comment by justinmk** [comment]
I think this can be closed.

**Comment by junegunn** [comment]
Right, thanks @justinmk.
